ipatests/test_integration/test_sudo.py

```diff
@@ -713,3 +713,42 @@ def test_category_all_validation_runasgroup(self):
                                           '--groups', 'testgroup2'],
                                           raiseonerr=False)
         assert result.returncode != 0
+
+    @pytest.mark.xfail(
+        reason="https://pagure.io/SSSD/sssd/issue/3957", strict=True)
+    def test_domain_resolution_order(self):
+        """Test sudo with runAsUser and domain resolution order.
+
+        Regression test for bug https://pagure.io/SSSD/sssd/issue/3957.
+        Running commands with sudo as specific user should succeed
+        when sudo rule has ipasudorunas field defined with value of that user
+        and domain-resolution-order is defined in ipa config.
+        """
+        self.master.run_command(
+            ['ipa', 'config-mod', '--domain-resolution-order',
+             self.domain.name])  # pylint: disable=no-member
+        try:
+            # prepare the sudo rule: set only one user for ipasudorunas
+            self.reset_rule_categories()
+            self.master.run_command(
+                ['ipa', 'sudorule-mod', 'testrule',
+                 '--runasgroupcat=', '--runasusercat='],
+                raiseonerr=False
+            )
+            self.master.run_command(
+                ['ipa', 'sudorule-add-runasuser', 'testrule',
+                 '--users', 'testuser2'])
+
+            # check that testuser1 is allowed to run commands as testuser2
+            # according to listing of allowed commands
+            result = self.list_sudo_commands('testuser1')
+            expected_rule = ('(testuser2@%s) NOPASSWD: ALL'
```

Comment:
The rule expects the qualified name but with the SSSD version used in PRCI, the non qualified name is returned:
```
transport.py 513 DEBUG RUN su -c "sudo -l -n" testuser1
transport.py 558 DEBUG Matching Defaults entries for testuser1@ipa.test on client0:
transport.py 558 DEBUG !visiblepw, env_reset, env_keep="COLORS DISPLAY HOSTNAME HISTSIZE KDEDIR LS_COLORS", env_keep+="MAIL PS1 PS2 QTDIR USERNAME LANG LC_ADDRESS LC_CTYPE", env_keep+="LC_COLLATE LC_IDENTIFICATION LC_MEASUREMENT LC_MESSAGES", env_keep+="LC_MONETARY LC_NAME LC_NUMERIC LC_PAPER LC_TELEPHONE", env_keep+="LC_TIME LC_ALL LANGUAGE LINGUAS _XKB_CHARSET XAUTHORITY", secure_path=/usr/local/sbin\:/usr/local/bin\:/usr/sbin\:/usr/bin\:/sbin\:/bin, !authenticate
transport.py 558 DEBUG
transport.py 558 DEBUG User testuser1@ipa.test may run the following commands on client0:
transport.py 558 DEBUG (testuser2) NOPASSWD: ALL
transport.py 217 DEBUG Exit code: 0
```
If you want to test the next command, you can relax the expected rule and use a matching reg exp instead.

Reply:
I consider this non-qualified name in sudo output to be another symptom of bug from issue 3954.